include/aspect/time_stepping/interface.h

```diff
@@ -0,0 +1,101 @@
+/*
+  Copyright (C) 2019 - 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#ifndef _aspect_time_stepping_interface_h
+#define _aspect_time_stepping_interface_h
+
+#include <aspect/global.h>
+#include <aspect/simulator_access.h>
+#include <aspect/termination_criteria/interface.h>
+
+namespace aspect
+{
+  using namespace dealii;
+
+  namespace TimeStepping
+  {
+
+    /**
+     * A class to handle computation of the next time step (as desired by the user) and
+     * checking if the simulation is finished.
+     */
+    template <int dim>
+    class Manager : public SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Override initialize_simulator() so that we can also initialize the contained
+         * termination_manager.
+         */
+        virtual void initialize_simulator (const Simulator<dim> &simulator_object) override;
+
+        /**
+         * Compute the size of the next time step potentially taking into account
+         * the current solution (convection time step, conduction time step), settings
+         * from parameters, and termination criteria (to hit the end time exactly).
+         */
+        double compute_time_step_size();
+
+        /**
+         * If true, the simulator should perform a checkpoint before terminating.
+         */
+        bool need_checkpoint_on_terminate();
+
+        /**
+         * Check if the simulation is ready to terminate sucessfully.
+         */
+        bool should_simulation_terminate_now();
```

Comment:
The name suggests that this function just answers the questions if the simulation should terminate, so could this be const too?